scex-core/src/main/scala/com/avsystem/scex/compiler/CodeGeneration.scala

```diff
@@ -178,14 +186,15 @@ object CodeGeneration {
 
           val typedVariables = variableTypes.toList.sorted.iterator.map {
             case (name, tpe) =>
+              val escapedName = escapeString(name)
               s"""
-                 |  private def ${name}VarTag = ${validatePrefix}inferVarTag[$tpe]$validatePostfix
+                 |  private def `${escapedName}VarTag` = ${validatePrefix}inferVarTag[$tpe]$validatePostfix
                  |
                  |  @$AnnotationPkg.NotValidated def `$name`: $tpe =
```

Comment:
This line and 196 too are the reasons why putting any of `IllegalCharsInVarName` into a var name will still fail - I didn't want to change it though because variables are read from and written to by accessing them by their name (see: `com.avsystem.scex.compiler.xmlfriendly.XmlFriendlyTranslator#variable`) - I didn't want to risk breaking the existing SCEX usages by applying escaping to these lines as well.

Reply:
Sure, good idea to avoid risk :+1: